igor-web/src/main/groovy/com/netflix/spinnaker/igor/build/BuildController.groovy

```diff
@@ -271,24 +262,27 @@ class BuildController {
         }
     }
 
-    private String getArtifactPathFromBuild(jenkinsService, job, buildNumber, String fileName) {
-        try {
-            return retrySupport.retry({ ->
-                def artifact = jenkinsService.getBuild(job, buildNumber).artifacts.find {
-                    it.fileName == fileName
-                }
-                if (artifact) {
-                    return artifact.relativePath
-                }
-                log.info("Could not find artifact {} for job {} on build #{}", fileName, job, buildNumber)
-                throw new ArtifactNotFoundException()
-            }, 5, 2000, false)
-        } catch (ArtifactNotFoundException ignored) {
-            return null
-        }
+    private String getArtifactPathFromBuild(jenkinsService, master, job, buildNumber, String fileName) {
+        return retrySupport.retry({ ->
+            def artifact = jenkinsService.getBuild(job, buildNumber).artifacts.find {
+                it.fileName == fileName
+            }
+            if (artifact) {
+                return artifact.relativePath
+            } else {
+                log.error("Unable to get igorProperties: Could not find build artifact matching requested filename '{}' on '{}' build '{}",
+                    kv("fileName", fileName), kv("master", master), kv("buildNumber", buildNumber))
+                throw new ArtifactNotFoundException(master, job, buildNumber, fileName)
+            }
+        }, 5, 2000, false)
     }
 
-    private static class ArtifactNotFoundException extends RuntimeException {}
+    @ResponseStatus(NOT_FOUND)
+    private static class ArtifactNotFoundException extends NotFoundException {
```

Comment:
Extending `NotFoundException` means we get Kork's default error handling & thus HTTP 404.